react-ui/src/ControlPanel.jsx

```diff
@@ -55,13 +60,28 @@ export class ControlPanel extends PureComponent {
     this.setState({ open: !this.state.open })
   }
 
-  setPictureFilter = ({ value }) => {
+  setPictureFilter = (value) => {
     this.props.setPictureFilter(value)
     this.props.captureUserPicture()
   }
+
+  setPictureFilterWeight = ({ target }) => {
+    this.props.setPictureFilterWeight(target.value)
+    this.props.captureUserPicture()
+  }
+
+  newPictureFilterFromImage = ({ target }) => {
+    const file = target.files[0]
+    imageDataFromFile(file,512)
+      .then(getImageStyle)
+      .then((value) => ({'label':file.name, value}))
+      .then(this.setPictureFilter)
+    target.value = '';
```

Comment:
Note that this approach goes against react way of doing things - we should not mutate dom in the outside like that.

This code does not seem to break even when this line is removed.